jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_routers.py

```diff
@@ -199,8 +211,12 @@ async def test_create_job_submission__with_execution_directory(
     create_data.pop("status", None)
     create_data.pop("client_id", None)
 
-    with time_frame() as window:
-        response = await client.post("/jobbergate/job-submissions/", json=create_data)
+    with mock.patch(
+        "jobbergate_api.apps.job_submissions.routers.get_job_properties_from_job_script"
+    ) as mocked:
+        mocked.return_value = JobProperties.parse_obj(create_data["execution_parameters"])
+        with time_frame() as window:
+            response = await client.post("/jobbergate/job-submissions/", json=create_data)
```

Comment:
why did you prefer to mock instead of "prepare" the test by adding a job script to the database previously?

Reply:
It was preferred following the encapsulation practice. The route has no need to know the implementation details from `get_job_properties_from_job_script` and neither does its test. In addition, there is no need to change this test if these internal implementation details change.

Reply:
hmmmmm, I don't think this is a good approach tbh. Let me think in some solid points then I update this thread

Reply:
so, you mock but at the end you don't assert if the mock has been actually called. Two things:

1. If you check the mock then you're testing something that shouldn't matter (because I can implement the same behaviour in a different way)
2. If you don't check the mock the it means you don't need to mock, right?

Reply:
That is a good catch, the assertion was missing, it was added in [2a11835](https://github.com/omnivector-solutions/jobbergate/pull/193/commits/2a11835048b6bfa416ec65d469d1733f5464cb2a).

Reply:
Here it is an example of what I have trying to mean:

```python
from datetime import datetime

@freeze("2022-12-05")
async def dummy_test():

    # manually insert here the application data
    # manually insert here the job script data

    payload = {
        "job_submission_name": "dummy",
        "job_submission_description": "dummy",
        "job_script_id": 1,
        "execution_directory": "/tmp",
        "client_id": "dummy",
        "execution_parameters": {
            "account": "dummy",
            "argv": "arg1 arg2 arg3 arg4",
            "environment": {"PATH": "/bin:/usr/bin/:/usr/local/bin/"},
            "partition": "compute",
        }
    }

    response = await client.post("/jobbergate/job-submissions/", json=payload)

    # assert whatever you want here

    # check if data is correctly in the database

    query = table.select().select(table.c.id == 1)
    record = await database.fetch_one(query)

    assert record.get("id") == 1 # this is the job submission ID. Check if it is 1 because the database is supposed to be empty
    assert record.get("job_submission_name") == "dummy"
    assert record.get("job_submission_description") == "dummy"
    assert record.get("job_submission_owner_email") == "whatever"
    assert record.get("job_script_id") == 1
    assert record.get("execution_directory") == "/tmp"
    assert record.get("slurm_job_id") is None
    assert record.get("client_id") == "dummy"
    assert record.get("status") == "CREATED"
    assert record.get("report_message") == str()
    assert record.get("created_at") == str(datetime.now())
    assert record.get("updated_at") == str(datetime.now())
    assert record.get("execution_parameters") == {
        "account": "dummy",
        "argv": "arg1 arg2 arg3 arg4",
        "environment": {"PATH": "/bin:/usr/bin/:/usr/local/bin/"},
        "partition": "compute",
    }
```

In this example you would really test if the endpoint works as expected: you input some data and expect this data to be in the database. Note that the code is agnostic on how you do it, so you could do by using the parser you built or whatever. There's no need to mock data.

Reply:
Note also that, there's no need to check if the variable `record` is None because, if it is, the test will fail and that's ok, it should fail indeed